source/swap/test/Swap.js

```diff
@@ -985,7 +1009,36 @@ contract('Swap', async accounts => {
       )
     })
 
-    it('Bob buys Ticket #12345 from Alice for 1 DAI', async () => {
+    it('Bob cannot Ticket #12345 from Alice if he sends id and amount in Party struct', async () => {
+      const order = await orders.getOrder({
+        signer: {
+          wallet: aliceAddress,
+          token: tokenTicket.address,
+          id: 12345,
+          amount: 100,
+          kind: ERC721_INTERFACE_ID,
+        },
+        sender: {
+          wallet: bobAddress,
+          token: tokenDAI.address,
+          amount: 100,
+        },
+      })
+
+      order.signature = await signatures.getWeb3Signature(
+        order,
+        aliceAddress,
+        swapAddress,
+        GANACHE_PROVIDER
+      )
+
+      await reverted(
+        swap(order, { from: bobAddress }),
+        'NO_AMOUNT_FIELD_IN_ERC721.'
```

Comment:
Why's there a full stop inside the revert message? Surely this test shouldnt pass with that 😢

Reply:
weird observation, if it's just a full stop, somehow truffle-assertions works, any other character, will fail it....

Reply:
That makes me sad. But good to know it still largely works.